iroh-p2p/src/node.rs

```diff
@@ -847,6 +962,50 @@ impl<KeyStorage: Storage> Node<KeyStorage> {
                     }
                 }
             }
+            RpcMessage::ListenForIdentify(response_channel, peer_id) => {
+                let channels = self.lookup_queries.entry(peer_id).or_default();
+                channels.push(response_channel);
+            }
+            RpcMessage::LookupPeerInfo(response_channel, peer_id) => {
+                if let Some(info) = self.swarm.behaviour().peer_manager.info_for_peer(&peer_id) {
+                    response_channel.send(info.last_info.clone()).ok();
+                } else {
+                    response_channel.send(None).ok();
+                }
+            }
```

Comment:
Not a big deal, but I would prefer if the calculation of what to send and the actual send was separated.